### DISCUSSION TITLE
msquic on Android/Xamarin

### DISCUSSION THREAD

**Discussion by deAtDamm** [microsoft/msquic#2329] [Q&A] [answered]
Hi

We would like to investigate the QUIC transport layer for our connection protocol between Android mobile and the backend.
Is it possible to download a pre-build android msquic library ?
Is there any C#/.NET library which is capable to use the msquic library on android ?

Thanks
Domenico

**Comment by thhous-msft** [comment, marked as answer]
We don't currently provide prebuilt android libraries. They are built in CI, and are easy to build in both WSL and on linux however. 
For C# wrappers, we have some very low level wrappers in the src/cs folder, but they require raw pointers and don't follow C# standards. There are some wrappers in .NET itself, but I'm not sure if they work in xamarin.

**Reply by ManickaP** [reply]
No support for Android, iOS or any other mobile platform in .NET runtime. We support Windows, Linux and Mac (I'm not 100% sure if we're fully there, @wfurt?)

**Comment by wfurt** [comment]
macOS should be close - there are two failing runtime test last time I checked. But AFIK we don't have CI test coverage. That should conceptually cover iOS and Android is generally Linux-is so the code should work IMHO without real code changes. .NET (HttpClient) will just light up if the library is there so this is mostly about availability of pre-builds, right?

**Reply by ManickaP** [reply]
System.Net.Quic is PNSE for anything other than Win/Lin/Mac and all H/3 gets trimmed from S.N.Http on the "unsupported" platforms.
